### ISSUE TITLE
Possible to use external dependencies for freetype2, bzip2 and zlib ?

### ISSUE THREAD

**Issue by sfroberg** [TurboVNC/turbovnc#111]
Hi!

Is it possible to use already installed, system-wide freetype2, bzip2 and zlib libraries
when compiling under Linux ?

Or is there some turbovnc specific changes to those libraries?

**Comment by dcommander** [comment]
The only TurboVNC-specific changes to those libs are cosmetic-- eliminating compiler warnings with `-Wall`, primarily.  There is no technical reason why TurboVNC couldn't be built against the system libraries.  I would just need to modify the build system such that it allows users to optionally do that.  Stand by.

**Comment by dcommander** [comment]
Implemented in both master and dev branches.  In addition to the feature you requested (which can be enabled by passing `-DTVNC_SYSTEMLIBS=1` to CMake), I also implemented a CMake option (`TVNC_SYSTEMX11`) that will cause the build to use the system-supplied X11 headers/libs.  For instance, on RHEL 6, I can generate a mostly system-specific build of TurboVNC like so:

```
cmake -G"Unix Makefiles" -DTVNC_SYSTEMLIBS=1 -DTVNC_SYSTEMX11=1 -DTVNC_DLOPENSSL=0
```

("mostly" = still requires the official libjpeg-turbo SDK, since RHEL 6 doesn't supply the TurboJPEG API library.)

On RHEL 7 & Fedora, I can generate a fully system-specific build of TurboVNC like so:

```
cmake -G"Unix Makefiles" -DTVNC_SYSTEMLIBS=1 -DTVNC_SYSTEMX11=1 -DTVNC_DLOPENSSL=0 -DTJPEG_INCLUDE_DIR=/usr/include -DTJPEG_LIBRARY=-lturbojpeg
```

Generally `TVNC_SYSTEMX11` won't work unless the system is using the same version of X.org as the TurboVNC Server, or a newer version.

**Comment by dcommander** [comment]
NOTE: If anyone who stumbles upon this issue is wondering why we can't also use the system-supplied X.org source, as TigerVNC does, that's unfortunately not feasible, given the number of customizations that TurboVNC makes to the X.org source.  Since TurboVNC is a cross-platform product, using an in-tree X.org source base ensures that I can maintain a consistent level of quality and performance across all platforms.  It also allows me to apply critical X.org bug fixes directly to the TurboVNC Server rather than relying on O/S vendors to do so.  It also allows me to use CMake to build the X server rather than trying to interface our CMake build system with the autotools build system used by X.org, and it allow the VNC server code to focus on only one X.org version rather than trying to adapt to all of them.

In short, `TVNC_SYSTEMX11` and `TVNC_SYSTEMLIBS` represents the limit of what is possible in terms of building a system-specific TurboVNC Server.

**Comment by sfroberg** [comment]
Thank you so much!
:-)
